### PR TITLE
Dev docker

### LINKED ISSUE #11: Create a dockerfile and also a Docker image for fund-my-watcard v0.2.0.
Dockerfile and Docker image should based on Ubuntu:16.04.

The version of Python should be larger than or equals to 3.5.

Also, some required packages should also be installed to ensure that everything work properly: chrome, vim, ... (Please take an investigation of it).

Thanks!

### PR DESCRIPTION
**Pull request by faushine**
Create a Dockerfile.
Hub: https://hub.docker.com/r/faushine/fund-watcard
Resolve #11

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    libncursesw5-dev \
+    libgdbm-dev \
+    libc6-dev \
+    zlib1g-dev \
+    libsqlite3-dev \
+    libssl-dev \
+    openssl \
+    libffi-dev
+
+WORKDIR /usr/local/share/
```
Comment (xingweitian) [inline Dockerfile]: Why we need to download chromedriver by hand?

Check [here](https://github.com/xingweitian/fund-my-watcard/blob/master/fund_my_watcard/watcard#L71), I use `webdriver_manager` to download the corresponding driver for Chrome.

So in my opinion, we actually don't need to download a specific chromedriver.

Also, I didn't see where are the commands to install Chrome, have you tried the docker image? This version of the dockerfile maybe cannot work well. [Here](https://www.itzgeek.com/how-tos/linux/ubuntu-how-tos/install-google-chrome-on-ubuntu-16-04.html) should be useful for you.

**Comment by xingweitian** [comment]
Thanks for making this PR! I just left some comments. Also, please indicate the link of the image on dockerhub in the README.md.

File: Dockerfile (hunk 1 above)
Comment (xingweitian) [reply Dockerfile]: In addition, let's think about if it is better to install fund-my-watcard from source code, or from pip.

**Comment by faushine** [comment]
Add webdriver options and revise Dockerfile.

**Comment by faushine** [comment]
Docker Hub: https://cloud.docker.com/u/faushine/repository/docker/faushine/fund-my-watcard

File: fund_my_watcard/watcard [hunk 2]
```diff
@@ -68,8 +69,11 @@ class MyWatCard:
 
     def add_fund(self, amount):
         try:
-            with Browser("chrome", **{"executable_path": ChromeDriverManager().install()}, incognito=True,
-                         headless=True) as browser:
+            chrome_options = webdriver.ChromeOptions()
+            chrome_options.add_argument('--headless')
+            chrome_options.add_argument('--no-sandbox')
```
Comment (xingweitian) [inline fund_my_watcard/watcard]: Why this option is necessary for us?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    && cd /usr/local/bin \
+    && ln -s /usr/bin/python3 python \
+    && pip3 install --upgrade pip \
+    && pip install fund-my-watcard
+
+# Commands to install Chrome and chromedriver
+RUN curl https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb -o /chrome.deb \
+    && dpkg -i /chrome.deb || apt-get install -yf \
+    && rm /chrome.deb \
+    && curl https://chromedriver.storage.googleapis.com/2.31/chromedriver_linux64.zip -o /usr/local/bin/chromedriver \
```
Comment (xingweitian) [inline Dockerfile]: Why we still need to download and configure chromedriver manually?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+FROM ubuntu
+
+LABEL maintainer="fyuxin@uwaterloo.ca"
+
+# Commands to install python
+RUN apt-get update \
+    && apt-get install -y python3-pip python3-dev vim curl apt-utils \
+    && cd /usr/local/bin \
+    && ln -s /usr/bin/python3 python \
+    && pip3 install --upgrade pip \
```
Comment (xingweitian) [inline Dockerfile]: Please unify `pip3` and `pip`. Choose one of them to use.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    && chmod +x /usr/local/bin/chromedriver
+
+# Commands to add source code to container for test
+# ADD ./ /usr/local/share/fund_my_watcard
+
+# Commands to add config file to container for test
+# ADD .watcard_config /root/.watcard_config
+
+# Commands to add config file to container
+ADD .watcard_config ~/.watcard_config
```
Comment (xingweitian) [inline Dockerfile]: Please add a new blank line at the end of the file.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    && pip install fund-my-watcard
+
+# Commands to install Chrome and chromedriver
+RUN curl https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb -o /chrome.deb \
+    && dpkg -i /chrome.deb || apt-get install -yf \
+    && rm /chrome.deb \
+    && curl https://chromedriver.storage.googleapis.com/2.31/chromedriver_linux64.zip -o /usr/local/bin/chromedriver \
+    && chmod +x /usr/local/bin/chromedriver
+
+# Commands to add source code to container for test
```
Comment (xingweitian) [inline Dockerfile]: Please remove all the useless comments.

**Comment by xingweitian** [comment]
> Docker Hub: https://cloud.docker.com/u/faushine/repository/docker/faushine/fund-my-watcard

Please mention it in README.md. That is to say, add it to README.md and commit it into this PR.

File: fund_my_watcard/watcard (hunk 2 above)
Comment (faushine) [reply fund_my_watcard/watcard]: Please read: https://sites.google.com/a/chromium.org/chromedriver/help/chrome-doesn-t-start

File: Dockerfile (hunk 3 above)
Comment (faushine) [reply Dockerfile]: Removed.

File: Dockerfile (hunk 4 above)
Comment (faushine) [reply Dockerfile]: Removed.

File: Dockerfile (hunk 5 above)
Comment (faushine) [reply Dockerfile]: Added.

File: Dockerfile (hunk 6 above)
Comment (faushine) [reply Dockerfile]: Removed.

**Comment by faushine** [comment]
> > Docker Hub: https://cloud.docker.com/u/faushine/repository/docker/faushine/fund-my-watcard
> 
> Please mention it in README.md. That is to say, add it to README.md and commit it into this PR.

Added.

**Comment by xingweitian** [comment]
> > > Docker Hub: https://cloud.docker.com/u/faushine/repository/docker/faushine/fund-my-watcard
> > 
> > 
> > Please mention it in README.md. That is to say, add it to README.md and commit it into this PR.
> 
> Added.

Is this link right? Or should it be `https://hub.docker.com/r/faushine/fund-watcard` ?

**Comment by xingweitian** [comment]
We actually don't need to use --no-sandbox option, we just need to set a proper user in container.

https://github.com/GoogleChromeLabs/lighthousebot/blob/master/builder/Dockerfile#L35-L40

**Comment by faushine** [comment]
Changing role not works since -no-sandbox is required for Linux. For example: https://github.com/GoogleChromeLabs/lighthousebot/blob/40fb90bb5b64731d552af423cf4f1bbdefe8eeed/builder/chromeuser-script_nonheadless.sh
They set "--disable-setuid-sandbox" when running chrome. I guess it is the same as "--no-sandbox".   The Docker hub should be https://hub.docker.com/r/faushine/fund-my-watcard. README has been updated.

**Pull request merged**